app/controllers/adoption_application_controller.rb

```diff
@@ -0,0 +1,29 @@
+class AdoptionApplicationController < ApplicationController
+  def new
+    @favorites = favorite.contents.keys
+  end
+
+  def create
+    new_application = AdoptionApplication.new(application_params)
+    if new_application.save
+      PetApplication.create(pet_id: params[:pets], adoption_application_id: new_application.id)
+      favorite.contents.delete(params[:pets])
```

Comment:
This deletes a pet from favorites if an application for that pet has been submitted. Let me know if you think there is a better way to accomplish this!

Reply:
Makes sense to me!